### ISSUE TITLE
Resolve coverage reporting issues with CodeCov

### ISSUE THREAD

**Issue by jywarren** [publiclab/plots2#6290]
After #5931, @kaustubh-nair opened #5954 to switch our test coverage reporting to CodeCov. But there is some issue with coverage esp with the separately threaded tests we have running now. We'd love help debugging/resolving this if anyone has some experience with Travis and coverage monitoring!

**Comment by jywarren** [comment]
Ideally we'll be starting to get a nice coverage graph like this! 

![image](https://user-images.githubusercontent.com/24359/64885138-550a7100-d631-11e9-8d3a-ca4ba1f22605.png)


In each PR. Of course, ideally it'd all be green 😄

**Comment by SidharthBansal** [comment]
@jywarren @sagarpreet-chadha can we create some minor projects from the above list as hall of fame tasks for GCI?

**Comment by SidharthBansal** [comment]
@jywarren i think making this as a whole as a hall of fame task is a great idea
(Maybe I am wrong. I think this task is removing code coverage errors from whole plots2 repo)

**Comment by jywarren** [comment]
Yes this'd be great!

**Comment by SidharthBansal** [comment]
Published

**Comment by Uzay-G** [comment]
Hey, I might work on this but was wondering, how can I test my changes to the code?
Thanks!

**Comment by Uzay-G** [comment]
I noticed that for some reason the config file has a section `omment` instead of `comment`. This seems a bit trivial but maybe that is a cause of the problem.
Also would anyone happen to know when the bot stopped working? That way I could try and pinpoint the problem.

**Comment by Uzay-G** [comment]
Also does the codecov config need to be in the master branch for the reports to work?
I am just trying things out that might work, sorry if I am making gross mistakes :smile:

**Comment by SidharthBansal** [comment]
@jywarren @Uzay-G do we need some more changes?
If there are only 3 lines of code change, I think this should be hard issue. If more work is needed then we can stick with HoF

**Comment by SidharthBansal** [comment]
Thanks Uzay-G for your help

**Comment by Uzay-G** [comment]
Thans @SidharthBansal. Yeah you can turn it into a hard one instead :+1:

**Comment by Uzay-G** [comment]
Now we just need to resolve the conflicts of #5954 and merge into master.

**Comment by SidharthBansal** [comment]
Please work on any other issue in the meantime. Jeff is on holidays.

**Comment by jywarren** [comment]
Just referencing a few issues together here. A recent example issue at https://github.com/publiclab/plots2/pull/6875#issuecomment-570288690

Issues possibly related to parallelization documented at https://github.com/publiclab/plots2/issues/5931

And follow-up notes and concerns from this process at #5954 ...

**Comment by jywarren** [comment]
In https://github.com/publiclab/plots2/pull/6789, coverage was to drop to 50%, but on rebasing, it went up to very little change and all passed. This makes me worry about the parallelization issue?

**Comment by jywarren** [comment]
OK, coming back here; i think things are OK... notes:

https://docs.codecov.io/docs/merging-reports says it can handle parallel reports already with no config needed.

https://docs.codecov.io/docs/unexpected-coverage-changes has some useful info too.

https://github.com/publiclab/plots2/pull/7061 was compaining that the diff coverage is not as high as the project (32% vs 80%) so I wonder @Uzay-G if we should give 1% flexibility on this, and how we configure this?

**Comment by jywarren** [comment]
Maybe we can lower the strictness here? https://docs.codecov.io/docs/codecovyml-reference#section-codecov

Also we can ignore some folders if we think that's appropriate...

**Comment by jywarren** [comment]
Yes, we can set threshold here! https://docs.codecov.io/docs/commit-status#section-threshold

**Comment by Uzay-G** [comment]
Yeah I think we need to make it more flexible. I will look at this later tonight when I can! :+1:

**Comment by Uzay-G** [comment]
Ok I can make a pull request for the threshold, do you think we should set it around 5-8%?

**Comment by jywarren** [comment]
Oh, i was thinking more like 1% - isn't it the "wiggle room" for the
overall project coverage to change +/- as a result of the PR? But yes,
let's try it out!

**Comment by Uzay-G** [comment]
Oh ok yeah alright! I'll make the pull request now for the threshold! :+1:
